website/docs/plugin/framework/migrating/resources/state-upgrade.mdx

```diff
@@ -0,0 +1,207 @@
+---
+page_title: 'Resources - State Upgrading: Migrating from SDKv2 to the Framework'
+description: >-
+  State upgraders let users update resources provisioned with old schema configurations.
+  Migrate resource StateUpgraders in SDKv2 to UpgradeState in the plugin Framework.
+---
+
+# State Upgraders
+
+When you update the resource implementation in your provider, some changes may not be compatible with old versions. You
+can create state upgraders to help users migrate resources provisioned with old schema configurations. Refer to
+[State Upgrade](/plugin/framework/resources/state-upgrade) in the Framework documentation for details.
+
+This page explains how to migrate resource `StateUpgraders` in SDKv2 to `UpgradeState` in the plugin Framework.
+
+## SDKv2
+
+In SDKv2, state upgraders are defined by populating the `StateUpgraders` field on the `schema.Resource` struct. Refer
+to [State Migration](/plugin/sdkv2/resources/state-migration) in the SDKv2 documentation for details.
+
+The following code shows a basic implementation of the `stateUpgraders` field in SDKv2. For clarity, the rest of the
+`schema.Resource` struct is omitted (denoted by `...`).
+
+```go
+func resourceExample() *schema.Resource {
+	return &schema.Resource{
+		StateUpgraders: []schema.StateUpgrader{
+			{
+				Version: int,
+				Type:    cty.Type,
+				Upgrade: StateUpgradeFunc,
+			},
+			...
+```
+
+## Framework
+
+In the Framework, you implement the `ResourceWithUpgradeState` on your `resource.Resource` type to upgrade your
+resource's state when required.
+
+The following code shows how you define an `UpgradeState` function with the Framework. The body of the `UpgradeState`
+function is omitted for clarity (denoted by `...`).
+
+```go
+func (r *resourceExample) UpgradeState(context.Context) map[int64]resource.StateUpgrader {
+	return map[int64]resource.StateUpgrader{
+		0: {
+			PriorSchema:  *tfsdk.Schema,
+			StateUpgrader: func(context.Context, UpgradeStateRequest, *UpgradeStateResponse),
```

Comment:
```suggestion
			StateUpgrader: upgradeExampleStateV0toV2(context.Context, UpgradeStateRequest, *UpgradeStateResponse),
```

Is this more correct/clear? I don't think we've used `func()` in this way elsewhere in the doc...

Reply:
I believe this is correct as this follows the [definition of the StateUpgrader field on the StateUpgrader struct](https://github.com/hashicorp/terraform-plugin-framework/blob/main/resource/state_upgrader.go#L36) in the framework. The difference between this example and, say, the example [StateUpgradeFunc](https://github.com/hashicorp/terraform-plugin-framework/pull/461/files#diff-4e8f4f6cd740e1ee36ef7492851e8364a2de2b03f1a2b6fe33d4616b2bd4d4ecR31) is that `StateUpgradeFunc` is defined as a type. It should be clear to Go developers I think.